LibreHardwareMonitorLib/Interop/NvidiaML.cs

```diff
@@ -21,10 +21,17 @@ internal static class NvidiaML
         private static WindowsNvmlDelegate _windowsNvmlInit;
         private static WindowsNvmlDelegate _windowsNvmlShutdown;
 
-        private static readonly IntPtr WindowsDll;
+        private static IntPtr WindowsDll;
 
-        static NvidiaML()
+        internal static bool IsAvailable { get; private set; }
+
+        internal static bool Initialize()
         {
+            if (IsAvailable)
+            {
+                throw new InvalidOperationException($"{nameof(NvidiaML)} is already initialized");
```

Comment:
What about just returning `true` here? Just in case multiple threads create an `NvidiaGpu`, although that's probably not very likely. It would then at least be able to continue with the program.

Reply:
Yeah, that is a better solution. I refactored this part and also made Initialize() and Close() methods threadsafe